cpp/oneapi/dal/graph/test/service_functions_test.cpp

```diff
@@ -0,0 +1,187 @@
+    std::array<std::int64_t, neighbors_count> neighbors = {1, 2, 4, 0, 0, 3, 6,
+                                                           2, 5, 7, 0, 3, 2, 3};
+  };
+
+  class complete_graph_type : public graph_base_data {
+  public:
+    complete_graph_type() {
+      filename = "test_complete_graph.csv";
+      file_content = "0 1 0 2 0 3 0 4 1 2 1 3 1 4 2 3 2 4 3 4";
+      vertex_count = 5;
+      edge_count = 10;
+    }
+
+    static const std::int64_t correct_vertex_count = 5;
+    static const std::int64_t neighbors_count = 20;
+    std::array<std::int64_t, correct_vertex_count> degrees = {4, 4, 4, 4, 4};
+    std::array<std::int64_t, neighbors_count> neighbors = {
+        1, 2, 3, 4, 0, 2, 3, 4, 0, 1, 3, 4, 0, 1, 2, 4, 0, 1, 2, 3};
+  };
+
+  class service_functions_test {
+  public:
+    using my_graph_type = dal::preview::undirected_adjacency_vector_graph<>;
+
+    template <typename GraphType> auto create_graph() {
+      GraphType graph_data;
+      graph_data.write_test_data();
+      const dal::preview::graph_csv_data_source ds(graph_data.get_filename());
+      const dal::preview::load_graph::descriptor<
+          dal::preview::edge_list<int32_t>, my_graph_type> desc;
+      auto my_graph = dal::preview::load_graph::load(desc, ds);
+      graph_data.delete_test_data();
+      return my_graph;
+    }
+
+    template <typename GraphType>
+    void check_vertex_degree(const GraphType &graph,
+                             std::int64_t *correct_degrees,
+                             std::int64_t vertex_count) {
+      int correct_degree_num = 0;
+      for (std::int64_t i = 0; i < vertex_count; i++) {
+        if (dal::preview::get_vertex_degree(graph, i) == correct_degrees[i])
+          correct_degree_num++;
+      }
+      REQUIRE(correct_degree_num == vertex_count);
+    }
+
+    template <typename GraphType>
+    void check_vertex_neighbors(const GraphType &graph,
+                                std::int64_t *correct_neighbors,
+                                std::int64_t vertex_count,
+                                std::int64_t neighbors_count) {
+      int correct_neighbors_num = 0;
+      std::int64_t neighbor_index = 0;
+      for (std::int64_t j = 0; j < vertex_count; j++) {
+        const auto[ start, end ] = dal::preview::get_vertex_neighbors(graph, j);
+        for (auto i = start; i != end; ++i) {
+          if (*i == correct_neighbors[neighbor_index])
+            correct_neighbors_num++;
+          neighbor_index++;
+        }
+      }
+      REQUIRE(correct_neighbors_num == neighbors_count);
+    }
+
+    void general_checks() {
+      two_vertices_graph_type tvg;
+      acyclic_graph_type ag;
+      complete_graph_type cg;
+
+      const auto two_vertices_graph = create_graph<two_vertices_graph_type>();
+      const auto acyclic_graph = create_graph<acyclic_graph_type>();
+      const auto complete_graph = create_graph<complete_graph_type>();
+
+      SECTION("check get_vertex_count") {

```

Comment:
Minor, can be modified later. I see here a pattern for creating loop cycle. What if you will extend number of graphs to check up to 15. 15 times the same code with different graph? You can use some container to store graphs and iterate over it using only 2 lines of code instead of 2 * num_graphs.